Added poco to glossary, and update refs - linking to this instead of Wiki (#27944)
dotnet/docs: docs/standard/glossary.md

"Platform" has a different meaning in the phrase "developer platform," which refers to software that provides tools and libraries for building and running apps. .NET is a cross-platform, open-source developer platform for building many different types of applications.

> ## poco

Reviewer: I think poco should be POCO everywhere in this section.

Author: Even in the heading?

> ## poco

Reviewer 2: Given that this is an acronym, I would capitalize it.
```suggestion
## POCO
```

> A poco&mdash;or a plain old class/[CLR](#clr) object&mdash;is a C# object that contains only public properties or fields. A poco shouldn't contain any other members, such as:

Reviewer 2: Outside of this wikipedia article, I've always heard it explained as "CLR object", which makes more sense to me than "class object", especially given the etymology indicating it came from Plain Old Java Object.

Also, I believe it's OK to refer to POCOs of .NET languages other than C#, which would be another reason to prefer CLR rather than "class".

```suggestion
A POCO&mdash;or a plain old [CLR](#clr) object&mdash;is an object that contains only public properties or fields. A POCO shouldn't contain any other members, such as:
```

Author: Before I saw the Wiki I'd never heard of it being called CLR object. I left it as both, but I've learned it as "class object".

> A poco&mdash;or a plain old class/[CLR](#clr) object&mdash;is a C# object that contains only public properties or fields. A poco shouldn't contain any other members, such as:

Author: ```suggestion
A POCO&mdash;or a plain old class/[CLR](#clr) object&mdash;is a C# object that contains only public properties or fields. A POCO shouldn't contain any other members, such as:
```

> A POCO&mdash;or a plain old class/[CLR](#clr) object&mdash;is a C# object that contains only public properties or fields. A POCO shouldn't contain any other members, such as:

Author: ```suggestion
A POCO&mdash;or a plain old class/[CLR](#clr) object&mdash;is a .NET data structure that contains only public properties or fields. A POCO shouldn't contain any other members, such as:
```

- methods
- events
- delegates

>  These objects are used primary as data transfer objects (DTOs). A pure _poco_ will not inherit another object, or implement an interface. It's common for pocos to be used with serialization.

Reviewer: ```suggestion
These objects are used primary as data transfer objects (DTOs). A pure POCO does not inherit another class or implement an interface. It's common for POCOs to be used with serialization.
```

>  These objects are used primary as data transfer objects (DTOs). A pure _poco_ will not inherit another object, or implement an interface. It's common for pocos to be used with serialization.

Reviewer 2: ```suggestion
 These objects are used primarily as data transfer objects (DTOs). A pure _POCO_ will not inherit another object, or implement an interface. It's common for POCOs to be used with serialization.
```

> These objects are used primary as data transfer objects (DTOs). A pure POCO does not inherit another class or implement an interface. It's common for POCOs to be used with serialization.

Author: ```suggestion
These objects are used primarily as data transfer objects (DTOs). A pure _POCO_ will not inherit another object, or implement an interface. It's common for POCOs to be used with serialization.
```
